stdlib/public/Concurrency/Task.swift

```diff
@@ -375,38 +392,26 @@ extension Task {
 
 }
 
-/// Run given throwing `operation` as part of a new top-level task.
-///
-/// Creating detached tasks should, generally, be avoided in favor of using
-/// `async` functions, `async let` declarations and `await` expressions - as
-/// those benefit from structured, bounded concurrency which is easier to reason
-/// about, as well as automatically inheriting the parent tasks priority,
-/// task-local storage, deadlines, as well as being cancelled automatically
-/// when their parent task is cancelled. Detached tasks do not get any of those
-/// benefits, and thus should only be used when an operation is impossible to
-/// be modelled with child tasks.
+/// Run the given non-throwing operation as part of a new top-level task.
 ///
-/// ### Cancellation
-/// A detached task always runs to completion unless it is explicitly cancelled.
-/// Specifically, dropping a detached tasks `Task.Handle` does _not_ automatically
-/// cancel given task.
+/// Avoid using a detached task unless it isn't possible
+/// to model the operation using structured concurrency features like child tasks.
+/// Child tasks inherit the parent task's priority,
+/// task-local storage, and deadlines,
+/// and canceling a parent task automatically cancels all of its child tasks.
+/// You need to handle these considerations manually with a detached task.
 ///
-/// Cancelling a task must be performed explicitly via `handle.cancel()`.
-///
-/// - Note: it is generally preferable to use child tasks rather than detached
-///   tasks. Child tasks automatically carry priorities, task-local state,
-///   deadlines and have other benefits resulting from the structured
-///   concurrency concepts that they model. Consider using detached tasks only
-///   when strictly necessary and impossible to model operations otherwise.
+/// A detached task runs to completion
+/// unless it is explicitly canceled by calling the `Task.Handle.cancel()` method.
+/// Specifically, dropping a detached task's handle
+/// doesn't cancel that task.
```

Comment:
This wording is tricky; the task _always_ runs to completion, but the cancellation may cause it to return early or throw etc.

But since cancellation is cooperative, this means that if the task never checks cancellation, it'd finish as if nothing ever cancelled it (this is by design).

So we should not say that cancellation prevents running to completion, we can rather say that it impacts the task and how it will return.

Reply:
Another note: cancellation does not prevent the task from being scheduled either, so it really is just informational to the task